utils/conditions/conditions.go

```diff
@@ -86,6 +86,15 @@ const (
 	// estimated once a rollout is paused.
 	RolloutPausedMessage = "Rollout is paused"
 
+	// RolloutBecomesIncomplete is added when a fully promoted rollout becomes incomplete, e.g.,
+	// due to  pod restarts, evicted -> recreated. In this case, we'll need to reset the rollout's
+	// condition to `PROGRESSING` to avoid any timeouts.
+	RolloutBecomesIncomplete = "RolloutBecomesIncomplete"
+	// RolloutBecomesIncompleteMessage is added when a fully promoted rollout becomes incomplete, e.g.,
+	// due to  pod restarts, evicted -> recreated. In this case, we'll need to reset the rollout's
+	// condition to `PROGRESSING` to avoid any timeouts.
+	RolloutBecomesIncompleteMessage = "Fully promoted rollout becomes incomplete"
+
```

Comment:
Lets create a new reason:

```golang
ReplicaSetNotAvailableReason = "ReplicaSet is not available"
```

and instead of `RolloutBecomesIncompleteMessage`, we can use existing `NotAvailableMessage` message